ddp-workspace/projects/ddp-sdk/src/lib/services/activity/activitySuggestionBuilder.service.ts

```diff
@@ -107,52 +121,32 @@ export class ActivitySuggestionBuilder {
             }
         };
 
-        let suggestionsLeft = suggestions.filter(hasAnyMatch);
-
-        const startsWithSuggestions = suggestionsLeft.reduce((accumulator: Array<TextSuggestion>, suggestion: string) => {
```

Comment:
I abstracted out these reductions to a `findMatches` closure since they're so similar.